rest-api.md

```diff
@@ -0,0 +1,767 @@
+}
+
+class NotFoundException implements HttpException {
+  statusCode: number = 404;
+}
+```
+
+Example:
+
+```
+curl -X 'POST' \
+  'http://localhost:3000/v1/modules/1/keys/find' \
+  -H 'Content-Type: application/json' \
+  -d '{
+  "pubkeys": [
+    "pubkey"
+  ]
+}'
+```
+
+:::warning
+If API returns 425 code, it means database is not ready for work
+:::
+
+### /validators
+
+**GET** `/v1/modules/{module_id}/validators/validator-exits-to-prepare/{operator_id}`
+
+`module_id` - staking router module contact address or id;
+
+This endpoint will return N of oldest lido validators for earliest epoch when voluntary exit can be processed for specific node operator and specific `StakingRouter` module. Node operator will use this validators list for preparing pre-sign exit messages. API will find `used` keys of node operator and find for these public keys N oldest validators in `Validator` table. We consider active validators (`active_ongoing` status) or validators in `pending_initialized`, `pending_queued` statuses. Module tables state fetched from `EL` should be newer than `Validator` table state fetched from `CL`. Otherwise API will return error on request.
+
+Query:
+
+Only one filter is available. If both parameters are provided, `percent` has a high priority.
+
+- `percent` - Percent of validators to exit. Default value is 10.
+- `max_amount` - Number of validators to exit. If validators number less than amount, return all validators.
+
+```typescript
+interface Validator {
+  validatorIndex: number;
+  key: string;
+}
+
+interface CLBlockSnapshot {
+  epoch: number;
+  root: number;
+  slot: number;
+  blockNumber: number;
+  timestamp: number;
+  blockHash: string;
+}
+
+class Response {
+  data: Validator[];
+  meta: {
+    clBlockSnapshot: CLBlockSnapshot;
+  };
+}
+
+interface HttpException {
+  statusCode: number;
+  message: string;
+}
+
+class TooEarlyResponse implements HttpException {
+  statusCode: number = 425;
+  message: string = 'Too early response';
+}
+
+class NotFoundException implements HttpException {
+  statusCode: number = 404;
+}
+
+class InternalServerErrorExceptionNotActualData implements HttpException {
+  statusCode: number = 500;
+  message: string = 'Last Execution Layer block number in our database older than last Consensus Layer';
+}
+
+class InternalServerErrorExceptionDisable implements HttpException {
+  statusCode: number = 500;
+  message: string = 'Validators Registry is disabled. Check environment variables';
+}
+```
+
+Example:
+
+```
+curl -X 'GET' 'http://localhost:3000/v1/modules/1/validators/validator-exits-to-prepare/1?percent=10'
+```
+
+:::warning
+If API returns 425 code, it means database is not ready for work
+:::
+
+**GET** `/v1/modules/{module_id}/validators/generate-unsigned-exit-messages/{operator_id}`
+
+`module_id` - staking router module contact address or id;
+
+Return unsigned exit messages for N oldest validators for earliest epoch when voluntary exit can be processed.
```

Comment:
```suggestion
Returns unsigned exit messages for validators sorded by activation_epoch?.
```

Reply:
i think better to highlight number. default is 10%

Reply:
and this `for earliest epoch when voluntary exit can be processed.` also important

Reply:
`N oldest` looks weird.  
Returns unsigned exit messages for validators sorted by activation_epoch that are eligible to exit.